psi-sample/client/src/main/java/com/r3/conclave/sample/client/Client.java

```diff
@@ -2,130 +2,148 @@
-import com.r3.conclave.common.EnclaveInstanceInfo;
-import com.r3.conclave.mail.Curve25519PrivateKey;
+import com.r3.conclave.client.EnclaveClient;
+import com.r3.conclave.client.web.WebEnclaveTransport;
+import com.r3.conclave.common.EnclaveConstraint;
 import com.r3.conclave.mail.EnclaveMail;
-import com.r3.conclave.mail.PostOffice;
 import com.r3.conclave.sample.common.AdDetails;
 import com.r3.conclave.sample.common.InputData;
 import com.r3.conclave.sample.common.InputDataSerializer;
 import com.r3.conclave.sample.common.UserDetails;
+import picocli.CommandLine;
 
-import java.io.*;
-import java.net.InetAddress;
-import java.net.InetSocketAddress;
-import java.net.Socket;
-import java.security.PrivateKey;
+import java.io.ByteArrayOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.UUID;
+import java.util.concurrent.Callable;
 
 /**
  * Client can be of two types: Merchant or Service Provider
  * Merchant - who supplies list of users who have made a purchase
  * Service Providers - who supplies a list of users who have clicked on the ad
  * Both the clients send the lists to enclave, which calculates the ad conversion rate and sends it back to the clients
  */
-public class Client {
+@CommandLine.Command(name = "psi-client",
+        mixinStandardHelpOptions = true,
+        description = "Simple client that communicates with the PSIEnclave using the web host.")
+public class Client implements Callable<Void> {
     private static final String MERCHANT = "MERCHANT";
     private static final String SERVICE_PROVIDER = "SERVICE-PROVIDER";
 
-    public static void main(String[] args) throws InterruptedException, IOException, InvalidEnclaveException {
+    //use picocli to provide command line parameters
+    @CommandLine.Parameters( description = "List of credit card numbers to be sent to enclave")
+        private List<String> creditCardNumbers = new ArrayList<String>();
 
-        if (args.length == 0) {
-            System.err.println("Please pass [MERCHANT/SERVICE-PROVIDER] [CONSTRAINT] followed by list of credit card numbers separated by spaces");
-            return;
+    @CommandLine.Option(names = {"-u", "--url"},
+                required = true,
+                description = "URL of the web host running the enclave.")
+        private String url;
+
+        @CommandLine.Option(names = {"-c", "--constraint"},
+                required = true,
+                description = "Enclave constraint which determines the enclave's identity and whether it's acceptable to use.",
+                converter = EnclaveConstraintConverter.class)
+        private EnclaveConstraint constraint;
+
+        @CommandLine.Option(names = {"-f", "--file-state"},
+                required = true,
+                description = "File to store the state of the client. If the file doesn't exist a new one will be created.")
+        private Path file;
```

Comment:
As far as I can see, your sample doesn't need a client state file. We can discuss offline.

Reply:
this sample does not require state save. hence removing the file option from command line and the file check in client code.